plugins/auth-backend-module-microsoft-provider/src/authenticator.ts

```diff
@@ -25,8 +26,16 @@ import { union } from 'lodash';
 
 /** @public */
 export const microsoftAuthenticator = createOAuthAuthenticator({
-  defaultProfileTransform:
-    PassportOAuthAuthenticatorHelper.defaultProfileTransform,
+  defaultProfileTransform: (
+    result: OAuthAuthenticatorResult<PassportProfile>,
+    context,
+  ) => {
+    result.fullProfile = result.fullProfile ?? {};
```

Comment:
Yep I think it could be good to move this fix here, as you mentioned: plugins/auth-node/src/oauth/PassportOAuthAuthenticatorHelper.ts

Don't really see a reason to not apply this across the board, at least making sure that it doesn't crash and just passes through empty info instead